draft-gazdag-x509-slhdsa.md

```diff
@@ -43,40 +54,434 @@ author:
+-->
+
+# Algorithm Identifiers {#sec-alg-ids}
+
+This specification uses placeholders for object identifiers until the identifiers for the new algorithms are assigned by NIST.
+
+The AlgorithmIdentifier type, which is included herein for convenience, is defined as follows:
+
+~~~
+    AlgorithmIdentifier  ::=  SEQUENCE  {
+        algorithm   OBJECT IDENTIFIER,
+        parameters  ANY DEFINED BY algorithm OPTIONAL
+    }
+
+    |  NOTE: The above syntax is from [RFC5280] and matches the
+    |  version used therein, i.e., the 1988 ASN.1 syntax.  See
+    |  [RFC5912] for ASN.1 copmatible with the 2015 ASN.1 syntax.
+~~~
+
+The fields in AlgorithmIdentifier have the following meanings:
+
+* algorithm identifies the cryptographic algorithm with an object identifier.
+
+* parameters, which are optional, are the associated parameters for the algorithm identifier in the algorithm field.
+
+The OIDs are:
+
+~~~
+    id-alg-slh-dsa-128s-shake OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+
+    id-alg-slh-dsa-128f-shake OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+
+    id-alg-slh-dsa-128s-sha2 OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+
+    id-alg-slh-dsa-128f-sha2 OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+
+    id-alg-slh-dsa-192s-shake OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+
+    id-alg-slh-dsa-192f-shake OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+
+    id-alg-slh-dsa-256s-shake OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+
+    id-alg-slh-dsa-256f-shake OBJECT IDENTIFIER ::= {
+            joint-iso-itu-t(2) country(16) us(840) organization(1)
+            gov(101) csor(3) nistAlgorithm(4) sigAlgs(3) TBD }
+~~~
+
+The contents of the parameters component for each algorithm are absent.
+
+# SLH-DSA Signatures in PKIX
+
+SLH-DSA is a digital signature scheme built upon hash functions. The security of SLH-DSA relies on the presumed diffculty of finding preimages for hash functions as well as several related properties of the same hash functions.
```

Comment:
My optinion: This paragraph belongs to the section "SLH-DSA Hash-based Signature Algorithm Overview"